File: Maximum_Ascending_Subarray_Sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        
        int cur = nums[0];
        int sum = nums[0];
        int n = nums.length;
        
        for(int i = 1; i<n; i++) {
            
            if( nums[i-1] < nums[i]  )
                cur += nums[i];
            else {
                cur = nums[i];
            }
            
            sum = Math.max(sum, cur);
                            
        }
        
        return sum;        
    }
}

File: Maximum_Ascending_Subarray_Sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int max=INT_MIN;
        int sum=nums[0];
        
        for(int i=1;i<nums.size();i++)
        {
            if(nums[i]>nums[i-1])
                sum=sum+nums[i];
            else
            {
                if(sum>max)
                    max=sum;
                
                sum=nums[i];
            }
        }
        if(sum>max)
          max=sum;
        return max;
    }
};
